src/room_membership/join_room/v1.rs

```diff
@@ -0,0 +1,62 @@
+//! [POST /_synapse/admin/v1/join/:room_id_or_alias](https://github.com/element-hq/synapse/blob/master/docs/admin_api/room_membership.md)
+
+use ruma::{
+    api::{request, response, Metadata},
+    metadata, OwnedRoomId, OwnedRoomOrAliasId, OwnedUserId,
+};
+
+const METADATA: Metadata = metadata! {
+    method: POST,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/join/:room_id_or_alias",
+    }
+};
+
+#[request]
+pub struct Request {
+    /// Alias or ID of the room to join.
+    #[ruma_api(path)]
+    pub room_id_or_alias: OwnedRoomOrAliasId,
+
+    /// User to join the room.
+    pub user_id: OwnedUserId,
+}
+
+#[response]
+pub struct Response {
+    /// Room ID of the joined room.
+    pub room_id: OwnedRoomId,
+}
+
+impl Request {
+    /// Creates a new `Request` with the given room or alias ID and user id.
+    pub fn new(room_id_or_alias: OwnedRoomOrAliasId, user_id: OwnedUserId) -> Self {
+        Self { room_id_or_alias, user_id }
+    }
+}
+
+impl Response {
+    /// Creates a new `Response` with the given room id
+    pub fn new(room_id: OwnedRoomId) -> Self {
+        Self { room_id }
+    }
+}
+
+#[test]
+fn test_join_room() {
+    use ruma::{owned_room_id, owned_user_id, OwnedRoomId, OwnedRoomOrAliasId, OwnedUserId};
+
+    let room_id: OwnedRoomId = owned_room_id!("!test:example.com");
+    let user_id: OwnedUserId = owned_user_id!("@carl:example.com");
+
+    // Check create request
+    let request = Request::new(room_id.clone().into(), user_id.clone());
+    assert_eq!(request.room_id_or_alias, OwnedRoomOrAliasId::from(room_id.clone()));
+    assert_eq!(request.user_id, user_id);
+
+    // Check create response
+    let response = Response::new(room_id.clone());
+    assert_eq!(response.room_id, room_id);
+}
```

Comment:
I agree, this test is not really useful ^^
Let's remove it.

```suggestion

impl Response {
    /// Creates a new `Response` with the given room id
    pub fn new(room_id: OwnedRoomId) -> Self {
        Self { room_id }
    }
}

#[test]
fn test_join_room() {
    use ruma::{owned_room_id, owned_user_id, OwnedRoomId, OwnedRoomOrAliasId, OwnedUserId};

    let room_id: OwnedRoomId = owned_room_id!("!test:example.com");
    let user_id: OwnedUserId = owned_user_id!("@carl:example.com");

    // Check create request
    let request = Request::new(room_id.clone().into(), user_id.clone());
    assert_eq!(request.room_id_or_alias, OwnedRoomOrAliasId::from(room_id.clone()));
    assert_eq!(request.user_id, user_id);

    // Check create response
    let response = Response::new(room_id.clone());
    assert_eq!(response.room_id, room_id);
}
```

Reply:
Oh, you disabled the "let maintainers edit the PR" setting?

Reply:
I think it's an org setting :). But I actually already implemented your suggested change ;)

Reply:
Oops, no-op change. I was saying it can be removed because it's not really testing anything. It's still there in the PR (just this one test).